src/cache_hash.cpp

```diff
@@ -343,7 +343,8 @@ Error cache_hit(CacheHash *ch, Buf *out_digest) {
             return ErrorInvalidFormat;
         }
         Buf *this_path = buf_create_from_slice(file_path);
-        if (chf->path != nullptr && !buf_eql_buf(this_path, chf->path)) {
+        // XXX: on FreeBSD, chf->path is same text but with extra \0 at the end
```

Comment:
Gotta track this bug down. Figure out where chf->path is getting set, figure out how the buf len gets set incorrectly. Can't just put a bandage on the symptom.

Reply:
It's the `SplitIterator` leaving a '\0' in the last component. Would it be okay to add `|| byte == 0` to `SplitIterator_isSplitByte`?